core/src/types/address.rs

```diff
@@ -134,6 +136,8 @@ pub enum Address {
     Implicit(ImplicitAddress),
     /// An internal address represents a module with a native VP
     Internal(InternalAddress),
+    /// An foreign address is provided from other chains
+    Foreign(String),
```

Comment:
btw, since we're storing foreign addresses on-chain, it might also be possible to store denominations for them, if they are a token address. this would allow you to specify a denominated amount when calling `ibc-transfer`